pkg/daemon/ceph/mgr/init.go

```diff
@@ -38,30 +38,34 @@ var (
 )
 
 const (
+	// InitCommand is the `rook ceph` subcommand which will perform mgr initialization
+	InitCommand = "mgr-init"
+
 	cephmgr = "ceph-mgr"
 )
 
+// Config contains the necessary parameters Rook needs to know to set up a mgr for a Ceph cluster.
 type Config struct {
 	ClusterInfo *cephconfig.ClusterInfo
 	Name        string
 	Keyring     string
 }
 
-func Run(context *clusterd.Context, config *Config) error {
-	logger.Infof("Starting MGR %s with keyring %s", config.Name, config.Keyring)
+// Initialize generates configuration files for a Ceph mgr
+func Initialize(context *clusterd.Context, config *Config) error {
+	logger.Infof("Creating config for MGR %s with keyring %s", config.Name, config.Keyring)
+	config.ClusterInfo.Log(logger)
 	if err := generateConfigFiles(context, config); err != nil {
 		return fmt.Errorf("failed to generate mgr config files. %+v", err)
 	}
 
-	if err := startMgr(context, config); err != nil {
-		return fmt.Errorf("failed to run mgr. %+v", err)
-	}
+	util.WriteFileToLog(logger, cephconfig.DefaultConfigFilePath())
+	util.WriteFileToLog(logger, cephconfig.DefaultKeyringFilePath())
 
 	return nil
 }
 
 func generateConfigFiles(context *clusterd.Context, config *Config) error {
-
 	keyringPath := getMgrKeyringPath(context.ConfigDir, config.Name)
 	confDir := getMgrConfDir(context.ConfigDir, config.Name)
```

Comment:
@travisn, regarding this line an the one above: I considered changing these paths to be the default ceph config path `/etc/ceph/keyring` and `/etc/ceph/ceph.conf`. This would mean that `/var/lib/rook` would no longer be used, though. Do you think this change would be out of scope given that the daemon is already configured to use `/etc/ceph` for the configs (except the mgr config)?

Reply:
if the daemon is configured to use `/etc/ceph`, it seems good to update this as well. There are just two reasons I can think of where we need to use the `/var/lib/rook` config
- Operator: The operator needs to be able to connect to multiple clusters, so we can't use /etc/ceph
- OSDs in a directory: The config should be in the same directory that is mounted for storage, instead of /etc/ceph

Reply:
Leaving this as `/var/lib/rook/mgr-<name>` for now. This could be changed to `/var/lib/ceph/mgr-a` in the future, but since mon still uses `/var/lib/rook`, it's probably best to leave it as-is for consistency.